packages/flutter_tools/test/general.shard/generate_localizations_test.dart

```diff
@@ -1882,6 +2025,50 @@ import 'output-localization-file_en.dart' deferred as output-localization-file_e
       expect(localizationsFile, contains(intlImportDartCode));
     });
 
+    testWithoutContext('intl package import should be kept in subclass files when select is included', () {
+      const String selectMessageArb = '''
+{
+  "genderSelect": "{gender, select, female {She} male {He} other {they} }",
+  "@genderSelect": {
+    "description": "A select message",
+    "placeholders": {
+      "gender": {}
+    }
+  }
+}
+''';
+
+      const String selectMessageEsArb = '''
+{
+  "genderSelect": "{gender, select, female {ES - She} male {ES - He} other {ES - they} }"
+}
+''';
+
+      fs.currentDirectory.childDirectory('lib').childDirectory('l10n')..createSync(recursive: true)
+        ..childFile(defaultTemplateArbFileName).writeAsStringSync(selectMessageArb)
+        ..childFile('app_es.arb').writeAsStringSync(selectMessageEsArb);
+
+      try {
+        LocalizationsGenerator(
```

Comment:
You don't need to explicitly fail here, throwing an unhandled exception will already fail the test

Reply:
Thanks for your feedback, resolved in the other PR.